internal/engine/watchmanager_test.go

```diff
@@ -0,0 +1,148 @@
+package engine
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/windmilleng/tilt/internal/logger"
+	"github.com/windmilleng/tilt/internal/model"
+	"github.com/windmilleng/tilt/internal/ospath"
+	"github.com/windmilleng/tilt/internal/store"
+	"github.com/windmilleng/tilt/internal/testutils/tempdir"
+	"github.com/windmilleng/tilt/internal/watch"
+)
+
+func TestWatchesReappliedOnDockerIgnoreChange(t *testing.T) {
+	f := newWMFixture(t)
+	defer f.TearDown()
+
+	setManifestTarget := func(target model.DockerComposeTarget) {
+		m := model.Manifest{Name: "foo"}.WithDeployTarget(target)
+		mt := store.ManifestTarget{Manifest: m}
+		state := f.store.LockMutableStateForTesting()
+		state.UpsertManifestTarget(&mt)
+		state.WatchMounts = true
+		f.store.UnlockMutableState()
+		f.wm.OnChange(f.ctx, f.store)
+	}
+
+	target := model.DockerComposeTarget{Name: "foo"}
+	target.Mounts = []model.Mount{{LocalPath: f.Path()}}
+	setManifestTarget(target.WithIgnoredLocalDirectories([]string{f.JoinPath("bar")}))
+
+	setManifestTarget(target)
+
+	f.ChangeFile(t, "bar")
+	f.ChangeFile(t, "stop")
+
+	actions, err := f.ReadActionsUntil("stop")
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	var paths []string
+	for _, a := range actions {
+		if !assert.Equal(t, 1, len(a.files)) {
```

Comment:
same nit as above

Reply:
also this means something is weird with the test fixture, doesn't it? maybe note that in the error message.